facia/app/services/PressedPageService.scala

```diff
@@ -0,0 +1,28 @@
+package services
+
+import java.nio.ByteBuffer
+import boopickle.Default._
+import common.Logging
+import concurrent.{BlockingOperations, FutureSemaphore}
+import model.PressedPage
+import protocol.BinaryPressedPageProtocol
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class PressedPageService(blockingOperations: BlockingOperations)
+                        (implicit ec: ExecutionContext) extends Logging with BinaryPressedPageProtocol {
+
+  private val futureSemaphore = new FutureSemaphore(15)
```

Comment:
How do you find `15` is the right value?
Is this not dependant of your instance type and num. of processors?

Reply:
Performance testing locally, it seems okay to allow a slightly greater rate than the machine is capable of processing. I chose 15 since my local machine (2.5 GHz Intel Core i7 - more powerful than the c4.2xlarge) can process 8-9 requests per second. 

I think even for the small EC2 instances, 15 would provide the behaviour we're trying to achieve - to fail fast, serve a limited number of requests and not fail the health check with thousands of CPU intensive futures attempting to execute.

Reply:
15 is concurrent requests, if the instance can do 8-9 per second then the maximum latency would be 2 seconds or thereabouts if the full 15 came in together.  Setting it too low would mean in the event of a spike in requests, we'll only accept the first n, setting it too high would mean that fastly would time out and not cache the responses (currently 10 second timeout?) or the ELB (60 second).

the aim is to not try to serve too many requests that won't make it into the cache (>10s) but not too low that a spike will be kicked out

Of course trying to make a self optimising system that would  cover all eventualities would not end well, so there's something to be said for simplicity, but I would suggest increasing it a bit.  You could always consider alarming to send an email on 99 percentile latency >10s?